src/services/tags.service.ts

```diff
@@ -31,3 +32,17 @@ export const deleteTagById = async (id: string) => {
   await deleteById(id);
   return { succeed: true };
 };
+
+export const addTagsFromContent = (tagNames: Set<string>) => {
+  tagNames.forEach(async (name: string) => {
+    const findResult = await getByName(name);
+
+    if (findResult === null) {
+      await create({ name: name }, { initialIncrementStats: true });
+    } else {
+      await update(findResult._id, {
+        stats: { tweets: ++findResult.stats.tweets },
+      });
+    }
+  });
```

Comment:
You can optimize this algorithm:
1. Do attempt to update tag by name with new stats
2. If it's failed to update, create a new tag
3. If it's success you're done

In that way you don't need to make "find" request